### PR TITLE
[new formula] - c v0.07

### PR DESCRIPTION
**Pull request by ryanmjacobs**
Fresh attempt at adding c to homebrew.

@mikemcquaid Would you mind reviewing this PR? Thanks.

### CODE REVIEW

**Comment by mikemcquaid** [comment]
@ryanmjacobs Could you consider rolling these changes into #37403 so we can keep discussion in one place? Thanks!

**Comment by mikemcquaid** [comment]
Ah, sorry, I see you didn't submit that PR, my bad.

File: Library/Formula/c.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+class C < Formula
+  homepage "https://github.com/ryanmjacobs/c"
+  url "https://github.com/ryanmjacobs/c/archive/v0.07.tar.gz"
+  sha1 "c40044a2747b97a2f8b550c1d417bb03848fcd40"
+
+  depends_on "gnu-sed"
+
+  def install
+      bin.install "c"
```
Comment (mikemcquaid) [inline Library/Formula/c.rb]: Two character indentation, thanks.

File: Library/Formula/c.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+class C < Formula
+  homepage "https://github.com/ryanmjacobs/c"
+  url "https://github.com/ryanmjacobs/c/archive/v0.07.tar.gz"
+  sha1 "c40044a2747b97a2f8b550c1d417bb03848fcd40"
+
+  depends_on "gnu-sed"
```
Comment (mikemcquaid) [inline Library/Formula/c.rb]: It looked like the upstream commit made this not need `gnu-sed` any more? Is it possible for this to be made to work with BSD `sed` instead? Thanks!

File: Library/Formula/c.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+  def install
+      bin.install "c"
+  end
+
+  test do
+    Dir.chdir("tests") do
+      system "test.sh"
```
Comment (mikemcquaid) [inline Library/Formula/c.rb]: This won't work. See `cmake.rb` for an example of an application formula with a good test. Thanks!

File: Library/Formula/c.rb (hunk 2 above)
Comment (ryanmjacobs) [inline Library/Formula/c.rb]: Okay, I think it works with both GNU and BSD sed now. It for sure works with GNU, but I don't have access to a BSD system.

If you have a BSD system, could you try these commands?
```bash
$ cat > file.c
#!/usr/bin/c blah blah blah
^D
$ sed -i.bak '1!b;s/^#!/\/\/#!/' file.c
$ cat file.c
//#!/usr/bin/c blah blah blah
```

Comment (mikemcquaid) [inline Library/Formula/c.rb]: If you remove the dependency and add something similar to the `test do` block the CI will verify for you.

**Comment by ryanmjacobs** [comment]
I guess we'll focus on PR #37403.

**Pull request closed without merging**